hack/e2e-test.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+
+set -e
+
+component=hive
+TEST_IMAGE=$(eval "echo $IMAGE_FORMAT")
+component=installer
+INSTALLER_IMAGE=$(eval "echo $IMAGE_FORMAT")
+
+
+# Install Hive
+make deploy DEPLOY_IMAGE="${TEST_IMAGE}"
+
+CLUSTER_PROFILE_DIR=/tmp/cluster
+
+# Create a new cluster deployment
+export BASE_DOMAIN="origin-ci-int-aws.dev.rhcloud.com"
+export CLUSTER_NAME="$(oc get cluster.cluster.k8s.io -n openshift-cluster-api -o jsonpath='{ .items[].metadata.name }')-1"
+export SSH_PUB_KEY="$(cat ${CLUSTER_PROFILE_DIR}/ssh-publickey)"
+export PULL_SECRET="$(cat ${CLUSTER_PROFILE_DIR}/pull-secret)"
+export AWS_ACCESS_KEY_ID=$(cat ${AWS_SHARED_CREDENTIALS_FILE} | awk '/aws_access_key_id/ { print $3; exit; }')
+export AWS_SECRET_ACCESS_KEY=$(cat ${AWS_SHARED_CREDENTIALS_FILE} | awk '/aws_secret_access_key/ { print $3; exit; }')
+
+function teardown() {
+	echo "Deleting ClusterDeployment ${CLUSTER_NAME}"
+	oc delete clusterdeployment ${CLUSTER_NAME}
+}
+trap 'teardown' EXIT
+
+oc new-project cluster-test
+
+oc process -f config/templates/cluster-deployment.yaml \
+   CLUSTER_NAME="${CLUSTER_NAME}" \
+   SSH_KEY="${SSH_PUB_KEY}" \
+   PULL_SECRET="${PULL_SECRET}" \
+   AWS_ACCESS_KEY_ID="${AWS_ACCESS_KEY_ID}" \
+   AWS_SECRET_ACCESS_KEY="${AWS_SECRET_ACCESS_KEY}" \
+   BASE_DOMAIN="${BASE_DOMAIN}" \
+   INSTALLER_IMAGE="${INSTALLER_IMAGE}" \
+   OPENSHIFT_RELEASE_IMAGE="" \
+   TRY_INSTALL_ONCE="true" \
+   | oc apply -f -
+
+# Wait for the cluster deployment to be installed
+SRC_ROOT=$(git rev-parse --show-toplevel)
+go run "${SRC_ROOT}/contrib/cmd/waitforjob/main.go" "${CLUSTER-NAME}-install"
```

Comment:
Maybe not now, but it would be nice if we had a go utililty to run a full integration test suite, so fo all of this, spin up the deployment, wait, make sure it's installed, and then run a suite of our own integration tests, check that we got things out of admin kubeconfig, etc. We don't really have that integration angle covered and it would be fairly easy for us to re-use it locally if it assumed you were logged into a cluster already (i.e. our dev clusters)